### ISSUE TITLE
Fahrplan HTMLTable Bug

### ISSUE THREAD

**Issue by Mephisto090** [Zefau/ioBroker.jarvis#1915]
*Please note that you may restore an old layout via jarvis adapter configuration directly within ioBroker

**Prerequisites / Voraussetzungen**
- [v3.0.13] Which jarvis version are you running / Welche Version von jarvis nutzt du? **BITTE GENAUE ANGABE!**
- [ ] Fehler / Screenshot der Browser Konsole (F12), siehe [Fehleranalyse und -Meldung](https://github.com/Zefau/ioBroker.jarvis/wiki/de-Errors)

**Describe the bug / Beschreibe den Fehler**
A clear and concise description of what the bug / problem is.
Eine genaue Beschreibung des Fehlers inkl. Herleitung, um diesen Nachstellen zu können.

Bei der HTMLTable wird mir mit den untenstehenden Werten immer in der Tabelle angezeigt: "No data available".

```
<table width="100%">
	<tr>
		<th align="left" colspan="4">A Dorf - B Dorf</th>
	</tr>
	<tr>
		<th align="center">Abfahrt</th>
		<th align="center">Verspätung</th>
		<th align="center">Plattform</th>
		<th align="center">Ankunft</th>
		<th align="center">Verspätung</th>
		<th align="center">Plattform</th>
	</tr>
	<tr>
		<td><font color="#008000">20:26</font></td>
		<td><font color="#008000">0</font></td>
		<td><font color="#008000">null</font></td>
		<td><font color="#008000">20:42</font></td>
		<td><font color="#008000">0</font></td>
		<td><font color="#008000">24</font></td>
	</tr>
	<tr>
		<td>20:50</td>
		<td>0</td>
		<td><font color="#008000">null</font></td>
		<td>23:05</td>
		<td>0</td>
		<td><font color="#008000">24</font></td>
	</tr>
	<tr>
		<td>20:50</td>
		<td>0</td>
		<td><font color="#008000">null</font></td>
		<td>23:09</td>
		<td>0</td>
		<td><font color="#008000">24</font></td>
	</tr>
</table>
```

**Screenshots**
Sofern möglich, bitte Screenshots einfügen, um den Fehler bildlich zu erläutern.

____

**Please note that bug reports may be closed if the information above are not given!**

**Bitte beachte, dass Bug-Reports ggf. einfach geschlossen werden, wenn die oben genannten Kriterien nicht angemessen gegeben sind!**

**Comment by Zefau** [comment]
`<th align="left" colspan="4">A Dorf - B Dorf</th>` muss raus, da `th` als Spalte interpretiert wird und in diesem Fall dann eine Spalte mehr ist als die Daten hergeben.

Außerdem ist `Verspätung` und `Plattform` doppelt drin. Dies fixe ich mit v3.1, damit die Spalten dann automatisch umbenannt werden (also ein `(1)` rangehangen wird). Wenn du es hier aber manuell umbenennst, sollte es klappen.

**Comment by mcuiobroker** [comment]
Wird bei mir aber angezeigt.
https://mcuiobroker.gitbook.io/jarvis-infos/tipps/adapter-spezifisch/fahrplan#fahrplan

**Comment by Zefau** [comment]
Du nutzt StateHTML und in diesem Issue wurde HTMLTable benutzt.

**Comment by mcuiobroker** [comment]
Stimmt. Wäre für die Anzeige auch sinnvoller? Es ist ja ein HTML DP.

**Comment by Zefau** [comment]
HtmlTable ist für die Anzeige von HTML Tabellen `<table ..>` gedacht, insofern schon richtig.

**Comment by Mephisto090** [comment]
> `<th align="left" colspan="4">A Dorf - B Dorf</th>` muss raus, da `th` als Spalte interpretiert wird und in diesem Fall dann eine Spalte mehr ist als die Daten hergeben.

Wäre das nicht möglich in Version 3.1. zu fixen und er auf colspan "aufpasst"?

**Comment by Mephisto090** [comment]
So funktioniert es:

```
<table width="100%">
	<tr>
		<th align="center">Abfahrt</th>
		<th align="center">Verspätung</th>
		<th align="center">Plattform</th>
		<th align="center">Ankunft</th>
		<th align="center">Verspätung 1</th>
		<th align="center">Plattform 1</th>
	</tr>
	<tr>
		<td><font color="#008000">20:26</font></td>
		<td><font color="#008000">0</font></td>
		<td><font color="#008000">null</font></td>
		<td><font color="#008000">20:42</font></td>
		<td><font color="#008000">0</font></td>
		<td><font color="#008000">24</font></td>
	</tr>
	<tr>
		<td>20:50</td>
		<td>0</td>
		<td><font color="#008000">null</font></td>
		<td>23:05</td>
		<td>0</td>
		<td><font color="#008000">24</font></td>
	</tr>
	<tr>
		<td>20:50</td>
		<td>0</td>
		<td><font color="#008000">null</font></td>
		<td>23:09</td>
		<td>0</td>
		<td><font color="#008000">24</font></td>
	</tr>
</table>
```


Wann soll Version 3.1. veröfentlich werden?

**Comment by Mephisto090** [comment]
> > `<th align="left" colspan="4">A Dorf - B Dorf</th>` muss raus, da `th` als Spalte interpretiert wird und in diesem Fall dann eine Spalte mehr ist als die Daten hergeben.
> 
> Wäre das nicht möglich in Version 3.1. zu fixen und er auf colspan "aufpasst"?

immer wieder den Adapter JavaScript damit beauftragen der alles "sauber" macht ist auch unnötige Ressourcen Verschwendung :(

**Comment by Zefau** [comment]
> Wäre das nicht möglich in Version 3.1. zu fixen und er auf colspan "aufpasst"?

Ja

**Comment by Mephisto090** [comment]
Darf ich das jetzt so verstehen, dass Version 3.1. es können wird? :D

Wann soll Version 3.1. kommen?

**Comment by Zefau** [comment]
> Darf ich das jetzt so verstehen, dass Version 3.1. es können wird? :D

Ja

> Wann soll Version 3.1. kommen?

in GitHub bereits als alpha drin und zeitnah auch als beta.
Im Latest Repo in Q4 oder Q1.

**Comment by Zefau** [comment]
schau nochmal mit `v3.1.0-alpha.49`

**Comment by Mephisto090** [comment]
Wie kann ich den Adapter auf eine alpha bringen?

**Comment by Zefau** [comment]
Direkt von GitHub indizieren l installieren

**Comment by Mephisto090** [comment]
Gibt es dort irgendwo eine Anleitung?

**Comment by Zefau** [comment]
Schau mal im Wiki

**Comment by Mephisto090** [comment]
Bekomme jetzt mit StandardHTML vom Adapter ein Fahrplan in der alpha angezeigt

**Comment by Mephisto090** [comment]
Gibt es eine Möglichkeit, dass ich Werte mit "null" überschreiben kann, sodass er ein "-" oder eine "0" anzeigt und nicht "null"

**Comment by Zefau** [comment]
> Bekomme jetzt mit StandardHTML vom Adapter ein Fahrplan in der alpha angezeigt

Teste mit der aktuellen Alpha bitte nochmal mittels HtmlTable. Damit sollte es nun auch gehen.

> Gibt es eine Möglichkeit, dass ich Werte mit "null" überschreiben kann, sodass er ein "-" oder eine "0" anzeigt und nicht "null"

Meinst du in der Tabelle oder bei normalen Datenpunkt-Werten?
- Bei normalen Datenpunkt-Werten mit der Anzeige-Variante.
- In der Tabelle via Spaltenkonfiguration und dort den `format` Parameter, womit du Werte bearbeiten kannst, siehe https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x#definition-von-jsontable-greater-than-infos-siehe-link

**Comment by mcuiobroker** [comment]
Ja, aber mehr für Datumsänderungen, nicht direkt für Wertänderung. Stelle ich aber noch rein.

**Comment by Mephisto090** [comment]
> Ja, aber mehr für Datumsänderungen, https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format nicht direkt für Wertänderung. Stelle ich aber noch rein.




> > Bekomme jetzt mit StandardHTML vom Adapter ein Fahrplan in der alpha angezeigt
> 
> Teste mit der aktuellen Alpha bitte nochmal mittels HtmlTable. Damit sollte es nun auch gehen.
> 

Ja funktioniert perfekt. Top!

> * In der Tabelle via Spaltenkonfiguration und dort den `format` Parameter, womit du Werte bearbeiten kannst, siehe https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x#definition-von-jsontable-greater-than-infos-siehe-link

Na dann warte ich mal bis du es ins wiki aufgenommen hast. :)

**Comment by mcuiobroker** [comment]
Ist doch schon drin
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#beispiel-wert-in-spalte-ersetzen
